File: app/src/main/java/edu/gatech/phileckstrom/faradaycallshield/misc/BaseActivity.java


package edu.gatech.phileckstrom.faradaycallshield.misc;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;
import androidx.appcompat.app.AppCompatActivity;


public abstract class BaseActivity extends AppCompatActivity {

    public static void addFragmentToActivity (FragmentManager fragmentManager,
                                              Fragment fragment,
                                              int frameId,
                                              String tag) {

        FragmentTransaction transaction = fragmentManager.beginTransaction();
        transaction.replace(frameId, fragment, tag);
        transaction.commit();
    }
}

File: app/src/main/java/edu/gatech/phileckstrom/faradaycallshield/misc/BaseActivity.kt
package edu.gatech.phileckstrom.faradaycallshield.misc

import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import edu.gatech.phileckstrom.faradaycallshield.services.PermissionsPrompt
import edu.gatech.phileckstrom.faradaycallshield.services.SingularEvent

abstract class BaseActivity : AppCompatActivity() {
    companion object {
        fun addFragmentToActivity(fragmentManager: FragmentManager,
                                  fragment: Fragment?,
                                  frameId: Int,
                                  tag: String?) {
            val transaction = fragmentManager.beginTransaction()
            transaction.replace(frameId, fragment!!, tag)
            transaction.commit()
        }
    }

    val uiEvent = SingularEvent<PermissionsPrompt>()
}